coffee-frontend/src/components/Wrapper/Wrapper.js

```diff
@@ -58,12 +95,28 @@ function Wrapper() {
   const theme = useTheme();
   const [mobileOpen, setMobileOpen] = React.useState(false);
   const auth = React.useState(true);
+  const [currentUser, setCurrentUser] = useState('')
+  const [state, dispatch] = React.useReducer(reducer, initialState);
+  const history = useHistory();
+
+
+  useEffect(() => {
+    async function getUser() {
+        const user = await getCurrentUser()
+        dispatch({type: "GETUSER", 
```

Comment:
Will this be called regardless if the call fails, or the user is not logged in ?

Reply:
If this will be called if the call fails? 

It will be called if the user is not logger in, this logs them in if they have a valid token, or doesn't log them in if they don't have a valid token